src/Api/JsonRpc/Repository/TestExecutionRepository.php

```diff
@@ -23,7 +23,7 @@ public function __construct(\Graze\GuzzleHttp\JsonRpc\Client $client)
         ]);
     }
 
-    public function findByTestCaseIdAndTestRunId(int $testCaseId, int $testRunId): ?TestExecution
+    public function findByTestCaseIdAndTestRunId(int $testCaseId, int $testRunId): array
```

Comment:
This method seems redundant. If test cases do not exist they need to be created first and I think this is already handled. 

Then calling `TestRun.add_case()` will return executions for the same run/case pair if they already exist or create new ones. You don't need to explicitly search for them b/c BE will do this for you.

Reply:
That will simplified the logic.